runtime-sdk/src/modules/accounts/types.rs

```diff
@@ -4,12 +4,32 @@ use std::collections::BTreeMap;
 use num_traits::identities::Zero;
 use serde::{Deserialize, Serialize};
 
-use crate::types::{address::Address, token};
+use crate::{
+    types::{address::Address, token},
+    crypto::signature::PublicKey
+};
+
+pub struct MultisigSigner {
+    pub public_key: PublicKey,
+    pub weight: u64,
+}
+
+pub struct MultisigAddressSpec {
+    pub signers: Vec<MultisigSigner>,
+    pub threshold: u64,
+}
+
+pub enum FromSpec {
+    SignerZero,
+    MultisigAddress(MultisigAddressSpec),
+}
 
 /// Transfer call.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
 pub struct Transfer {
+    // todo: maybe skip serializing if SignerZero
+    pub from: FromSpec,
```

Comment:
Hm would be better if multisig would work for all transactions. You could make it a separate public key type that includes this spec?

Reply:
I last brought up this idea of treating multisig as a signature algorithm, conceptually next to ed25519 and others, when we were discussing multisig for oasis-core. but at that time we concluded that time that it was too weird. are there any new considerations now that would change our opinion on it?

Reply:
Don't remember what the reasoning was then for Oasis Core, but SDK already supports multiple different signature schemes  (Ed25519 and Secp256k1) and treating "Multisig v1" as another one makes sense to me.

Reply:
for now I'm going to move the types and stuff from modules/accounts out to the addresses module or something